### PR TITLE
Update Map.update example to use full lambda syntax on first example

### PR DESCRIPTION
**Pull request by olivermt**
Hello.

I was training a new person today and he was very confused on the &(&...) syntax for updating in Map.update.

I have changed the first example to use the full lambda syntax so that it is more clear that the argument should be a regular lambda with an output and that the value is set to the return value of this function.

### CODE REVIEW

File: lib/elixir/lib/map.ex [hunk 1]
```diff
@@ -604,7 +604,7 @@ defmodule Map do
 
   ## Examples
 
-      iex> Map.update(%{a: 1}, :a, 13, &(&1 * 2))
+      iex> Map.update(%{a: 1}, :a, 13, fn existing -> existing * 2 end)
```
Comment (eksperimental) [inline lib/elixir/lib/map.ex:607]:
```suggestion
      iex> Map.update(%{a: 1}, :a, 13, fn value -> value * 2 end)
```

Comment (eksperimental) [reply lib/elixir/lib/map.ex:607]: `value` reads better IMO

Comment (olivermt) [reply lib/elixir/lib/map.ex:607]: How about `existing_value`? There is nothing wrong with being extra verbose imo.

Comment (eksperimental) [reply lib/elixir/lib/map.ex:607]: Yes. But use `current_value`. That's what's being used in the example in `Map.get_and_update/3`

Comment (eksperimental) [reply lib/elixir/lib/map.ex:607]: and I would replace the captures in the second example as well.

Comment (josevalim) [inline lib/elixir/lib/map.ex:607]:
```suggestion
      iex> Map.update(%{a: 1}, :a, 13, fn existing_value -> existing_value * 2 end)
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**